midx.c

```diff
@@ -798,6 +820,12 @@ int write_midx_file(const char *object_dir)
 			packs.pack_name_concat_len += strlen(packs.names[packs.nr]) + 1;
 			packs.nr++;
 		}
+
+		if (packs_to_drop && (drop_index < packs_to_drop->nr || missing_drops)) {
+			error(_("did not see all pack-files to drop"));
+			result = 1;
+			goto cleanup;
```

Comment:
all of the "goto cleanup" statements will skip the close_midx() on 856.  Is this OK?

Reply:
We only need to close it if we are going to rewrite the file. Git normally doesn't close the midx at the end of the process, anyway.